experimental/dds/tree2/src/simple-tree/proxies.ts

```diff
@@ -718,13 +706,13 @@ interface ExtractedFactoryContent<T extends TypedNode<TreeNodeSchema, "javaScrip
  * }
  * ```
  */
-export function extractFactoryContent<T extends TypedNode<TreeNodeSchema, "javaScript">>(
+export function extractFactoryContent<T extends TypedNodeFactoryInput<TreeNodeSchema>>(
 	content: T,
 ): ExtractedFactoryContent<T> {
 	if (isFluidHandle(content)) {
 		return { content, hydrateProxies: noopHydrator };
-	} else if (Array.isArray(content)) {
-		return extractContentArray(content);
+	} else if (isReadonlyArray(content)) {
```

Comment:
since the arrays we care about are readonly, narrowing with Array.isArray just gives never if the isn't just `undefined`. This is why the line below compiled without the cast.

Reply:
Good catch, thanks.